ui/src/containers/AlertPage.js

```diff
@@ -179,260 +158,81 @@ export const TableHeader = styled.th`
-                    style: column.cellStyle,
-                  }),
-                );
-                return (
-                  <TableHeader {...headerStyleProps} className="th">
-                    {column.render('Header')}
-                    <SortCaretWrapper>
-                      {column.isSorted ? (
-                        column.isSortedDesc ? (
-                          <i className="fas fa-sort-down" />
-                        ) : (
-                          <i className="fas fa-sort-up" />
-                        )
-                      ) : (
-                        <SortIncentive>
-                          <i className="fas fa-sort" />
-                        </SortIncentive>
-                      )}
-                    </SortCaretWrapper>
-                  </TableHeader>
-                );
-              })}
-            </HeadRow>
-          );
-        })}
-      </thead>
-
-      <Body {...getTableBodyProps()}>
-        {!rows.length && (
-          <EmptyTable>
-            {intl.formatMessage({ id: 'no_active_alerts' })}
-          </EmptyTable>
-        )}
-        {rows.map((row, i) => {
-          prepareRow(row);
-          return (
-            <HeadRow {...row.getRowProps()}>
-              {row.cells.map((cell) => {
-                let cellProps = cell.getCellProps({
-                  style: {
-                    ...cell.column.cellStyle,
-                  },
-                });
-                if (cell.column.Header === 'Active since') {
-                  return (
-                    <td {...cellProps}>
-                      <span>{formatDateToMid1(cell.value)}</span>
-                    </td>
-                  );
-                } else if (cell.column.Header === 'Severity') {
-                  return (
-                    <td {...cellProps}>
-                      <CircleStatus status={cell.value} />
-                    </td>
-                  );
-                } else {
-                  return <td {...cellProps}>{cell.render('Cell')}</td>;
-                }
-              })}
-            </HeadRow>
-          );
-        })}
-      </Body>
-    </table>
-  );
-}
+        </div>
+        <Table.SingleSelectableContent
+          rowHeight="h64"
+          separationLineVariant="backgroundLevel1"
+          backgroundVariant="backgroundLevel2"
+          customItemKey={(index, data) => {
+            return data[index].id;
+          }}
+        />
+      </Table>
+    );
+  },
+  (a, b) => {
+    // compare the alert only on id and severity
+    return isEqual(a.columns, b.columns) && isEqualAlert(a.data, b.data);
+  },
+);
 
 export default function AlertPage() {
   const alerts = useAlerts({});
+
   const leafAlerts = useMemo(
     () => alerts?.alerts.filter((alert) => !alert.labels.children) || [],
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [JSON.stringify(alerts?.alerts)],
+    [alerts?.alerts],
```

Comment:
I think we should keep the previous code.

React uses “referential equality” for object and array within the dependencies array.
so even we have the same content of alerts, it will still re-run the effect. That's why we stringify. 

A pretty good article to read about.
https://www.benmvp.com/blog/object-array-dependencies-react-useEffect-hook/